### PR TITLE
fix(storage): refresh users list on each display in the add container view for object-storage and cold-archive.

### PR DESCRIPTION
**Pull request by brofik**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/MANAGER-11093` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-9964 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

The PR :https://github.com/ovh/manager/pull/8870  must be merged before this one.

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/storage-users.service.js [hunk 1]
```diff
@@ -41,6 +41,18 @@ export default class PciStoragesUsersService {
       .then(({ data }) => data);
   }
 
+  mapUsersToCredentials(projectId, users) {
+    const usersCredentialsPromises = users.map((user) =>
+      this.getS3Credentials(projectId, user.id).then((data) => {
+        const updatedUser = user;
+        [updatedUser.s3Credentials] = data;
+        return updatedUser;
+      }),
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/storage-users.service.js:50]: **nitpick:**  Unnecessary local variable
```suggestion
      this.getS3Credentials(projectId, user.id).then((data) => ({
         ...user,
         s3Credentials: data,
      })),
```

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/components/associate-user-to-container/controller.js [hunk 2]
```diff
@@ -36,21 +36,38 @@ export default class CreateLinkedUserController {
...
+            );
+          })
+          .map((user) => {
+            const updatedUser = user;
+            updatedUser.credentialTrad = this.getCredentialTranslation(user);
+            updatedUser.userNameDescriptionKey = user.description
+              ? `${user.username} - ${user.description}`
+              : user.username;
+            return updatedUser;
+          });
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/containers/add/components/associate-user-to-container/controller.js:64]: **nitpick:** Unnecessary local variable
```suggestion
          .map((user) => ({
            ...user,
            credentialTrad:this.getCredentialTranslation(user);
            userNameDescriptionKey: user.description
              ? `${user.username} - ${user.description}`
              : user.username,
          }));
```

File: packages/manager/modules/pci/src/projects/project/storages/storage-users.service.js (hunk 1 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/storage-users.service.js:50]: This re-writing works only if we want to add a new pair (key, value) to an existing object.
It doesn't work when we need to update the value of an object attribute.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js [hunk 3]
```diff
@@ -55,13 +54,31 @@ export default class ColdArchiveLinkUserArchiveController {
...
+      .then((usersWithCredentials) => {
+        this.usersCredentials = usersWithCredentials.map((user) => {
+          const updatedUser = user;
+          [updatedUser.s3Credentials] = user.s3Credentials;
+          updatedUser.credentialTrad = this.getCredentialTranslation(user);
+          updatedUser.userNameDescriptionKey = user.description
+            ? `${user.username} - ${user.description}`
+            : user.username;
+          return updatedUser;
+        });
```
[the remark this answers is not in the document]
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js:75]: same as https://github.com/ovh/manager/pull/8910#discussion_r1113723171

File: packages/manager/modules/pci/src/projects/project/storages/storage-users.service.js (hunk 1 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/storage-users.service.js:50]: Done.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js:75]: Done.

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/components/associate-user-to-container/controller.js (hunk 2 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/project/storages/containers/add/components/associate-user-to-container/controller.js:64]: Done.

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js [hunk 4]
```diff
@@ -55,18 +54,35 @@ export default class ColdArchiveLinkUserArchiveController {
...
+      .then((usersWithCredentials) => {
+        this.usersCredentials = usersWithCredentials.map((user) => ({
+          ...user,
+          s3Credentials: user?.s3Credentials[0],
+          credentialTrad: this.getCredentialTranslation(user),
+          userNameDescriptionKey: user.description
+            ? `${user.username} - ${user.description}`
+            : user.username,
+        }));
+        this.users = this.usersCredentials;
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js:75]: **question:** Why do you need same data on 2 different keys on the controller object? `this.users` and `this.usersCredentials`
```suggestion
        this.users = usersWithCredentials.map((user) => ({
          ...user,
          s3Credentials: user?.s3Credentials[0],
          credentialTrad: this.getCredentialTranslation(user),
          userNameDescriptionKey: user.description
            ? `${user.username} - ${user.description}`
            : user.username,
        }));
```

File: packages/manager/modules/pci/src/projects/project/storages/containers/add/components/associate-user-to-container/controller.js [hunk 5]
```diff
@@ -36,21 +36,37 @@ export default class CreateLinkedUserController {
...
+            );
+          })
+          .map((user) => ({
+            ...user,
+            credentialTrad: this.getCredentialTranslation(user),
+            userNameDescriptionKey: user.description
+              ? `${user.username} - ${user.description}`
+              : user.username,
+          }));
+        this.users = this.usersCredentials;
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/containers/add/components/associate-user-to-container/controller.js:64]: **nitpick:** Avoid having same data in multiple variables.
```suggestion
        this.users = usersWithCredentials
          .filter(({ roles }) => {
            return roles.find(({ name }) =>
              [
                USER_ROLES.ADMINISTRATOR,
                USER_ROLES.OBJECTSTORE_OPERATOR,
              ].includes(name),
            );
          })
          .map((user) => ({
            ...user,
            credentialTrad: this.getCredentialTranslation(user),
            userNameDescriptionKey: user.description
              ? `${user.username} - ${user.description}`
              : user.username,
          }));
```

**Review by anooparveti — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**